zk/src/org/zkoss/zk/ui/sys/ShadowElementsCtrl.java

```diff
@@ -128,12 +128,13 @@ private static void doBindChildrenInFilter(Component[] shadows) {
 				if (!se.getDistributedChildren().isEmpty() || !((ShadowElementCtrl) se).isDynamicValue())
 					continue;
 				List<Component> children = shadow.getChildren();
-				if (children.size() > 0) {
-					doBindChildrenInFilter(children.toArray(new Component[children.size()]));
-				} else {
+				if (shadow.getChildren().size() == 0) {

```

Comment:
duplicate to List<Component> children = shadow.getChildren();

Reply:
The status the shadow.getChildren() may change.